extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/http/CatalogResource.java

```diff
@@ -0,0 +1,586 @@
+        }
+        catch (DuplicateKeyException e) {
+          // Table exists
+          if (overwrite) {
+            // User wants to overwrite, so do so.
+            newVersion = catalog.tables().replace(table);
+          } else {
+            throw e;
+          }
+        }
+      }
+      return okWithVersion(newVersion);
+    }
+    catch (CatalogException e) {
+      return e.toResponse();
+    }
+  }
+
+  /**
+   * Retrieves the table metadata, including the spec.
+   * <p>
+   * Returns a 404 (NOT FOUND) error if the table catalog entry does not exist.
+   * Note that this check is only for the <i>specification</i>; the table (or
+   * datasource) itself may exist. Similarly, this call may return a specification
+   * even if there is no datasource of the same name (typically occurs when
+   * the definition is created before the datasource itself.)
+   *
+   * @param dbSchema The Druid schema. The user must have read access.
+   * @param name The name of the table within the schema. The user must have
+   *        read access.
+   * @param req the HTTP request used for authorization.
+   * @return the definition for the table, if any.
+   */
+  @GET
+  @Path("/schemas/{schema}/tables/{name}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getTable(
+      @PathParam("schema") String dbSchema,
+      @PathParam("name") String name,
+      @Context final HttpServletRequest req
+  )
+  {
+    try {
+      final SchemaSpec schema = validateSchema(dbSchema, false);
+      authorizeTable(schema, name, Action.READ, req);
+      final TableMetadata table = catalog.tables().read(new TableId(dbSchema, name));
+      return Response.ok().entity(table).build();
+    }
+    catch (CatalogException e) {
+      return e.toResponse();
+    }
+  }
+
+  /**
+   * Deletes the table definition (but not the underlying table or datasource)
+   * for the given schema and table.
+   *
+   * @param dbSchema The name of the schema that holds the table.
+   * @param name The name of the table definition to delete. The user must have
+   *             write access.
+   */
+  @DELETE
+  @Path("/schemas/{schema}/tables/{name}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response deleteTable(
+      @PathParam("schema") String dbSchema,
+      @PathParam("name") String name,
+      @Context final HttpServletRequest req
+  )
+  {
+    try {
+      final SchemaSpec schema = validateSchema(dbSchema, true);
+      authorizeTable(schema, name, Action.WRITE, req);
+      catalog.tables().delete(new TableId(dbSchema, name));
+      return ok();
+    }
+    catch (CatalogException e) {
+      return e.toResponse();
+    }
+  }
+
+  // ---------------------------------------------------------------------
+  // Modify a table within the catalog
+
+  @POST
+  @Path("/schemas/{schema}/tables/{name}/edit")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response editTable(
```

Comment:
would be good to add more details about the different edit operations here, and their semantics, or maybe just a pointer the the different edit classes, which themselves describe semantics.